rasa/shared/utils/common.py

```diff
@@ -5,32 +5,59 @@
 import logging
 from typing import Text, Dict, Optional, Any, List, Callable, Collection
 
+import rasa.shared.utils.io as io_utils
+from rasa.shared.constants import NEXT_MAJOR_VERSION_FOR_DEPRECATIONS
+
+
 logger = logging.getLogger(__name__)
 
 
 def class_from_module_path(
-    module_path: Text, lookup_path: Optional[Text] = None
+    module_path: Text, lookup_path: Optional[Text] = None, ensure_class: bool = True
 ) -> Any:
     """Given the module name and path of a class, tries to retrieve the class.
 
-    The loaded class can be used to instantiate new objects."""
-    # load the module, will raise ImportError if module cannot be loaded
+    The loaded class can be used to instantiate new objects.
+
+    Args:
+        module_path: either an absolute path to a Python class,
+                     or the name of the class in the local / global scope.
+        lookup_path: a path where to load the class from, if it cannot
+                     be found in the local / global scope.
+        ensure_class: a boolean that will ensure that the loaded Python class
+                      is actually class if True (by default).
+
+    Returns:
+        a Python class
+
+    Raises:
+        ImportError, in case the Python class cannot be found.
+    """
+    klass = None
     if "." in module_path:
         module_name, _, class_name = module_path.rpartition(".")
         m = importlib.import_module(module_name)
-        # get the class, will raise AttributeError if class cannot be found
-        return getattr(m, class_name)
+        klass = getattr(m, class_name, None)
     else:
-        module = globals().get(module_path, locals().get(module_path))
-        if module is not None:
-            return module
+        klass = globals().get(module_path, locals().get(module_path))
 
-        if lookup_path:
+        if klass is None and lookup_path:
             # last resort: try to import the class from the lookup path
             m = importlib.import_module(lookup_path)
-            return getattr(m, module_path)
-        else:
-            raise ImportError(f"Cannot retrieve class from path {module_path}.")
+            klass = getattr(m, module_path, None)
+
+    if klass is None:
+        raise ImportError(f"Cannot retrieve class from path {module_path}.")
+
+    if ensure_class and not inspect.isclass(klass):
+        io_utils.raise_deprecation_warning(
+            f"`class_from_module_path()` is expected to return a class, "
+            f"but {module_path} is not one. "
```

Comment:
what do you think? 
```suggestion
            f"but {module_path} is of type '{type(klass)}'. "
```

Reply:
🤔 I don't think it adds much. The path will be something like `path.to.Class`, I think there's enough information already, `is of type 'Class'` doesn't help.